src/hotspot/share/interpreter/bytecodeTracer.cpp

```diff
@@ -55,34 +55,32 @@ class BytecodePrinter {
   Bytecodes::Code _code;
   address   _next_pc;                // current decoding position
   int       _flags;
+  bool      _is_linked;
 
+  bool      is_linked() const        { return _is_linked; }
   void      align()                  { _next_pc = align_up(_next_pc, sizeof(jint)); }
   int       get_byte()               { return *(jbyte*) _next_pc++; }  // signed
   short     get_short()              { short i=Bytes::get_Java_u2(_next_pc); _next_pc+=2; return i; }
   int       get_int()                { int i=Bytes::get_Java_u4(_next_pc); _next_pc+=4; return i; }
 
   int       get_index_u1()           { return *(address)_next_pc++; }
-  int       get_index_u2()           { int i=Bytes::get_Java_u2(_next_pc); _next_pc+=2; return i; }
-  int       get_index_u1_cpcache()   { return get_index_u1() + ConstantPool::CPCACHE_INDEX_TAG; }
-  int       get_index_u2_cpcache()   { int i=Bytes::get_native_u2(_next_pc); _next_pc+=2; return i + ConstantPool::CPCACHE_INDEX_TAG; }
-  int       get_index_u4()           { int i=Bytes::get_native_u4(_next_pc); _next_pc+=4; return i; }
-  int       get_index_special()      { return (is_wide()) ? get_index_u2() : get_index_u1(); }
-  Method* method()                 { return _current_method; }
-  bool      is_wide()                { return _is_wide; }
-  Bytecodes::Code raw_code()         { return Bytecodes::Code(_code); }
-
-
-  bool      check_index(int i, int& cp_index, outputStream* st);
-  bool      check_cp_cache_index(int i, int& cp_index, outputStream* st);
-  bool      check_obj_index(int i, int& cp_index, outputStream* st);
-  bool      check_invokedynamic_index(int i, int& cp_index, outputStream* st);
+  int       get_native_index_u2()    { int i=Bytes::get_native_u2(_next_pc); _next_pc+=2; return i; }
+  int       get_native_index_u4()    { int i=Bytes::get_native_u4(_next_pc); _next_pc+=4; return i; }
+  int       get_Java_index_u2()      { int i=Bytes::get_Java_u2(_next_pc); _next_pc+=2; return i; }
+  int       get_Java_index_u4()      { int i=Bytes::get_Java_u4(_next_pc); _next_pc+=4; return i; }
```

Comment:
Since you've done the reformatting, can you add spaces around i=Bytes ? I don't really care if you add spaces around _next_pc+=x.  That looks ok to me.

Reply:
I reformatted the code and reordered the functions a little.